src/thoth/query/_engine.py

```
@@ -129,59 +106,46 @@ def answer(
-
-        Candidates are gathered in the existing stable order (folder order from
-        :data:`SEARCHED_DIRS`, then filename order) and **stable-sorted by that key
-        descending**, so two pages with an identical key keep that original order (the
-        pre-#96 tie-break). The ranked list is then deduplicated-by-construction and
-        capped at ``limit``.
+        """Lexically scans the searched folders for ``term``, ranked by hits.
+
+        Each page scores on how many distinct query tokens it matches, so a
+        natural-language query surfaces the page hitting the most words first even when
+        it lives in a folder scanned last (issue #96). Tokens match on word boundaries
+        (``<token>``), case-insensitively, so ``bed`` no longer matches ``embedded``
```

Ruff (0.16.2)

[error] Invalid unescaped character backspace, use "\b" instead

Replace with escape sequence

(PLE2510)